pgjdbc/src/main/java/org/postgresql/core/v3/ProtocolConnectionImpl.java

```diff
@@ -57,7 +57,7 @@ public String getServerVersion() {
     public int getServerVersionNum() {
         if (serverVersionNum != 0)
             return serverVersionNum;
-        return Utils.parseServerVersionStr(serverVersion);
+        return serverVersionNum = Utils.parseServerVersionStr(serverVersion);
```

Comment:
It's not thread safe in a sense that once updated every other thread will immediately see the new version. But that's not really important, since the other thread will just recalculate it again.
It's atomic (unlike long) so that is not an issue.
Or did you have something else in mind?